### ISSUE TITLE
win_msi troubles with Exit-Json and logcontent

### ISSUE THREAD

**Issue by davidobrien1985** [ansible/ansible-modules-core#2330]
Hi,

anybody come across the issue that win_msi just fails because of Exit-Json not being able to properly return the log content?

Got an MSI that created a 9kb log file, it takes Exit-Json about 5mins (on a decent VM) to convert that to Json and then fails with this message:

``` powershell
failed: [sv0000101np.domain.local] => {"failed": true, "parsed": false}
#< CLIXML
<Objs Version="1.1.0.1" xmlns="http://schemas.microsoft.com/powershell/2004/04"><S S="Error">ConvertTo-Json : Cannot convert value to type System.String._x000D__x000A_</S><S S="Error">At C:\Users\svc_go\AppData\Local\Temp\ansible-tmp-1445485201.78-60445853908721\_x000D__x000A_</S><S S="Error">win_msi.ps1:87 char:17_x000D__x000A_</S><S S="Error">+     echo $obj | ConvertTo-Json -Depth 99_x000D__x000A_</S><S S="Error">+                 ~~~~~~~~~~~~~~~~~~~~~~~~_x000D__x000A_</S><S S="Error">    + CategoryInfo          : InvalidArgument: (:) [ConvertTo-Json], PSInvalid _x000D__x000A_</S><S S="Error">   CastException_x000D__x000A_</S><S S="Error">    + FullyQualifiedErrorId : InvalidCastFromAnyTypeToString,Microsoft.PowerSh _x000D__x000A_</S><S S="Error">   ell.Commands.ConvertToJsonCommand_x000D__x000A_</S><S S="Error"> _x000D__x000A_</S></Objs>
```

Doesn't matter how large the log file is, in my tests it always fails.

It works fine with a straight string.

Seen that before?

**Comment by cchurch** [comment]
@davidobrien1985 Do you have a publicly available (i.e. one I can use to test) MSI that causes this problem?

**Comment by davidobrien1985** [comment]
Unfortunately, no. Also, I can only reproduce it in the customer's environment.
They are on WS2k8R2 with PowerShell version 4.

For now I am overwriting the module and not returning any log.

In my test environment, with the same MSI, the $result.log is null by the way, which is very odd.

**Comment by h0nIg** [comment]
reproduced with Pscx-3.2.0.msi , see https://pscx.codeplex.com/releases

**Comment by h0nIg** [comment]
i have to correct myself: this error will occur if the msi installation is short because 90% of the installations, the file is still empty after calling msiexec (async call?) and therefore the conversion will not happen. If you add this pull request and set wait=yes, the problem will be shown https://github.com/ansible/ansible-modules-core/pull/1109 , since its guaranteed that the file is filled

**Comment by h0nIg** [comment]
@sivel , could you please take a look?

**Comment by h0nIg** [comment]
i took a look and the reason is some weird stuff with ConvertTo-Json and the depth of 99. I would like to suggest the following PR, which removed the content attribute and instead gives the user the log file path. with this information, the user can still fetch the file and examine it

https://github.com/ansible/ansible-modules-core/pull/2563

**Comment by jhawkesworth** [comment]
I think it likely that this issue will be fixed by this PR: https://github.com/ansible/ansible-modules-core/pull/2563

**Comment by nitzmahone** [comment]
fix merged for 2.0.1 and devel
